### PR TITLE
Monitor directions #2

### PR DESCRIPTION
**Pull request by jeanpijon**
Modification of previous PR https://github.com/fchollet/keras/pull/823. 
Now follows standard API of monitor directions.

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -173,14 +173,29 @@ def on_epoch_end(self, epoch, logs={}):
...
+        
+        if mode == "min":
+            self.monitor_op = np.less
+            self.best = np.Inf
+        elif mode == "max":
+            self.monitor_op = np.greater
+            self.best = -np.Inf
+        else:
+            if mode != 'auto':
+                warnings.warn("ModelCheckpoint mode %s is unknown, fallback to auto mode" % (self.mode), RuntimeWarning)
```
Comment (fchollet) [inline keras/callbacks.py]: No warning should occur with default settings. It is safe to remove this.

File: keras/callbacks.py [hunk 2]
```diff
@@ -189,7 +204,7 @@ def on_epoch_end(self, epoch, logs={}):
             if current is None:
                 warnings.warn("Can save best model only with %s available, skipping." % (self.monitor), RuntimeWarning)
             else:
-                if current < self.best:
+                if self.monitor_op(current,self.best):
```
Comment (fchollet) [inline keras/callbacks.py]: Space after the comma.

File: keras/callbacks.py [hunk 3]
```diff
@@ -173,14 +173,29 @@ def on_epoch_end(self, epoch, logs={}):
...
+    def __init__(self, filepath, monitor='val_loss', verbose=0, save_best_only=False, mode='auto'):
 
         super(Callback, self).__init__()
         self.monitor = monitor
         self.verbose = verbose
         self.filepath = filepath
         self.save_best_only = save_best_only
-        self.best = np.Inf
+        
+        if mode == "min":
```
Comment (fchollet) [inline keras/callbacks.py]: Please initially check that `mode` in is `{'auto', 'min', 'max'}`.

**Comment by fchollet** [comment]
Ok, now checks are passing (we had an issue with Travis), and it's looking good.

Since this is a small change, could you squash the different commits into a single one with an appropriate description? This makes changes easier to follow in the commit history. Then I'll merge this.

**Comment by jeanpijon** [comment]
OK, done...

**Comment by fchollet** [comment]
Thanks! Merged.

**Pull request merged**